bepasty_cli/cli.py

```diff
@@ -55,10 +59,7 @@ def main(token, fileobj, fname, url, ftype):
         stdin = False
 
     else:
-        #  tried to write stdin in tempfile and run guess_file, but guess_file
-        #  only seem to look on the file extension ...
-        # tmpfile = NamedTemporaryFile(delete=False)
-        fileobj = click.get_binary_stream('stdin')
+        fileobj = BytesIO(click.get_binary_stream('stdin').read())
```

Comment:
reading the first 1024 bytes may be enough to detect the shebang but things like zip cannot be detected because their signature is at the end of the file.

the basic idea would be:

``` diff
-        fileobj = BytesIO(click.get_binary_stream('stdin').read()) 
+        mimeobj = BytesIO(click.get_binary_stream('stdin').read(1024)) 
...
```

another idea would be to read ahead the first 1024 bytes of the file which will be sent to the server and use it for mime detection.